#1104 Modify docs for background tasks

f0t0n opened the pull request:
<!-- Thank you for your contribution!

     Before submit your Pull Request, make sure you picked
     the right branch:

     - If you propose a new feature or improvement, select "master"
       as a target branch;
     - If this is a bug fix or documentation amendment, select
       the latest release branch (which looks like "0.xx") -->

## What do these changes do?

<!-- Please give a short brief about these changes. -->

Changes:

- Modify docs for background tasks
- Add an example code to examples directory

As discussed the old code example for background tasks creating was working but not optimal.
Removed `asyncio.gather()` from the example and used `app.loop.create_task()` instead.
Also added an example of tasks cancellation on cleanup.

## Are there changes in behavior for the user?

<!-- Outline any notable behaviour for the end users. -->
None

## Related issue number

<!-- Are there any issues opened that will be resolved by merging this change? -->
#1104

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

f0t0n commented:
@asvetlov Do you think it's short enough or I should remove something (https://github.com/KeepSafe/aiohttp/issues/1104#issuecomment-241540140)?

![image](https://cloud.githubusercontent.com/assets/957767/17870741/4fc4baa6-68c1-11e6-8732-01a922daba50.png)

asvetlov commented:
First.
I think `quickly_notify_monitoring` is not necessary.
If you want to inform users to allowance for doing an instant activity on app startup -- please invite something more obvious than just `sleep`.
I have troubles with a good example personally,

Second.
Your code uses both zmq and redis. For sake of shortness we should choose the single one.
I believe redis is more well known library, let's keep it.

Redis example may be more realistic than, again, just sleeps in a loop.
Create and redis connection using aioredis, subscribe for events and iterate over subscribed messages.

I gives to reader a clue for real use cases.

f0t0n commented:
@asvetlov Thanks for guidance. I've just pushed new changes and rebased the branch to lastest master.

Currently it looks like this:

![image](https://cloud.githubusercontent.com/assets/957767/17909245/529b912a-698c-11e6-827a-5651646271e7.png)

Also I've rewrote examples/background_tasks.py and ran it with redis to ensure it works and correctly. So the example in docs is an extract from there and is working fine.

asvetlov commented on examples/background_tasks.py:
```diff
@@ -0,0 +1,62 @@
+            for ws in app['websockets']:
+                ws.send_str('{}: {}'.format(ch.name, msg))
+            print("message in {}: {}".format(ch.name, msg))
+    except asyncio.CancelledError:
```
`finally` is better than `except CancelledError`.
We should close redis subscription unconditionally.

f0t0n replied:
Agreed. Gonna modify now.

f0t0n replied:
Done.

f0t0n commented:
@asvetlov Should I squash commits or keep for history?

asvetlov commented:
Doesn't matter -- I'm doing squashing on merge anyway.

asvetlov commented:
Perfect!
